atlp-rwanda/e-commerce-lydia-32-bn
# improved payment with stripe and ensured sending emails to sellers an…

### What does this PR do?
This PR improves the payment functionality by integrating Stripe for secure payment processing and order status updates.

### Description of Task to be completed?
1. Stripe Integration: Implemented Stripe payment gateway to facilitate seamless transactions.
2. Order Status Update: Automatically updates order status upon successful payment.
3. Email Notifications: Sends confirmation emails to both the buyer and product vendors upon successful order placement.

### How should this be manually tested?
1.Initiate Payment:

1.1 Navigate to the payment route( '/api/payment/pay') of the application.
1.2 Enter an order ID for payment in request params.
1.3 Verify that the Stripe payment gateway initializes correctly.
1.4 Enter returned payment session URL in browser to finish payment.

2. Proceed with payment success and order status update
   2.1 Navigate to the payment success route( '/api/payment/success') of the application.
   2.2 Enter payment session ID and order ID in request params
   2.3 If payment was successful order status will be updated to 'Paid' 

3. Cancel Payment Session
  3.1 Navigate to the payment cancelling route( '/api/payment/cancel') of the application.
  3.2 Enter payment session ID and order ID in request params
  3.3 If payment is still pending, it will be cancelled, otherwise you'll 'Payment not found' as response
